### ISSUE TITLE
[MCS Library] socket.connect() failed after uploading data points for several times.

### ISSUE THREAD

**Issue by dariachen** [MediaTek-Labs/Arduino-Add-On-for-LinkIt-SDK#13]
Upload integer, hex and string data points 13 times for each and got socket.connect() failed, socket.status()=0 in "bool MCSDevice::_uploadDataPoint(const String& params)" method.

**Comment by pablosun** [comment]
There are memory leaks. After several TCP connection being created and closed, the follow error log is emitted:

```
[Arduino:atcp_start:716] TTCP [0x2000b4b8]: could not allocate payload
```

By enabling the option `MTK_SUPPORT_HEAP_DEBUG = y` in `feature.mk` we are able to trace the leakage is caused by the `atcp_start` function:
```c
	if (ttcp->mode == TTCP_MODE_TRANSMIT) {		//for client

		int8_t id = insertNewClientConn(ttcp, p);	//always insert 0.
		if (id == NO_VALID_ID)
			return ERR_MEM;

		//Global val, Indicate client whether connect server success.
		_connected[ttcp->sock] = 0;

		ttcp->payload[id] = malloc(ttcp->buflen);	//malloc receive buf
```

It seems that `ttcp` is not mananged properly. Need further investigation to find where the leakage happens.

**Comment by pablosun** [comment]
should be fixed by version **v0.7.6-alpha**, which is available in the development board manager URL:
```txt
http://download.labs.mediatek.com/package_mtk_linkit_7697_dev_index.json
```
